disco_aws_automation/disco_elb.py

```diff
@@ -168,21 +168,25 @@ def get_or_create_target_group(self, group_name, vpc_id=None,
                 port = 80
                 health_protocol = 'HTTP'
                 health_port = '80'
-            if health_check_path:
-                health_check_path = health_check_path
+
+            health_check_path = health_check_path or '/'
+
+            health_args = dict()
+            if health_protocol in ('HTTP', 'HTTPS'):
+                health_args['HealthCheckProtocol'] = health_protocol
+                health_args['HealthCheckPort'] = health_port
+                health_args['HealthCheckPath'] = health_check_path
+
             else:
-                health_check_path = '/'
+                health_args['HealthCheckProtocol'] = "TCP"
```

Comment:
if the protocol is not http or https the only other protocol supported for a targetgroup is TCP which will handle all non http/s. 
```HealthCheckProtocol (string) -- The protocol the load balancer uses when performing health checks on targets. For Application Load Balancers, the default is HTTP. For Network Load Balancers, the default is TCP. The TCP protocol is supported for health checks only if the protocol of the target group is TCP, TLS, UDP, or TCP_UDP. The TLS, UDP, and TCP_UDP protocols are not supported for health checks.``` https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/elbv2.html#ElasticLoadBalancingv2.Client.create_target_group

Reply:
I believe it should also have `health_args['HealthCheckPort'] = health_port` for TCP, otherwise how it would know what TCP connection to check.

Reply:
it defaults to the traffic port but i missread the docs and thought that override was only for htttp, adding change now